src/main.c

```diff
@@ -86,7 +86,7 @@ make_conn(const char *hostname)
  * the message and return 1.
  */
 int
-send_request_and_read_response(int sockfd, void *client_session,
+send_request_and_read_response(int sockfd, rustls_client_session *client_session,
```

Comment:
nit: It's a shame that the Rust type doesn't materialize in C as `struct rustls_client_session` as I think that's what the [`curl` style would prefer](https://github.com/curl/curl/blob/master/docs/CODE_STYLE.md). What _does_ this type look like in the generated C header? Is it a typedef to `void *`?

Reply:
Here's what it looks like in the .h:

```
typedef struct rustls_client_session rustls_client_session;                                                                                                                                                        
```

So I probably could use the `curl` style. I'll give it a try.